tasks/linux.sh

```diff
@@ -2,61 +2,12 @@
 
 # example: bolt task run package::linux action=install name=rsyslog
 
-# $1 = string literal to feed to dpkg-query. include json formatting
-# No error check here because querying the status of uninstalled packages will return non-zero
-# Use --showformat to make a json object with status and version
-# ${name%%=*} removes the version in case we installed a specific one
-# This will print nothing to stdout if the package is not installed
-apt_status() {
-  dpkg-query --show --showformat="$1" "${name%%=*}"
-}
-
-# Determine if newer package is available to mirror the ruby/puppet implementation
-apt_check_latest() {
-  installed="$(apt_status '${Version}')"
-  [[ $installed ]] || success '{ "status": "uninstalled", "version": ""}'
-
-  candidate="$(apt-cache policy "$name" | grep 'Candidate:')"
-  candidate="${candidate#*: }"
-
-  if [[ $installed != $candidate ]]; then
-    cmd_status="$(apt_status "{ \"status\":\"\${Status}\", \"version\":\"${installed}\", \"latest\":\"${candidate}\" }")"
-  else
-    cmd_status="$(apt_status '{ "status":"${Status}", "version":"${Version}" }')"
-  fi
-
-  success "$cmd_status"
-}
-
-# $1 = string literal to feed to rpm -q. include json formatting
-# Use --queryformat to make a json object with status and version
-# yum returns non-zero if the package isn't installed
-yum_status() {
-  rpm -q --queryformat "$1" "$provided_package"
-}
-
-# Determine if newer package is available to mirror the ruby/puppet implementation
-yum_check_latest() {
-  installed="$(yum_status "%{VERSION}-%{RELEASE}" "$provided_package")" || success '{ "status": "uninstalled", "version": "" }'
-  candidate=($(yum check-update --quiet "${name}"))
-
-  # format of check-update is <package_name> <release>
-  # i.e rpm.x86_64 4.11.3-35.el7
-  if [[ $candidate && $installed != ${candidate[1]} ]]; then
-    cmd_status="$(yum_status \
-      "{ \"status\": \"installed\", \"version\": \"$installed\", \"latest\": \"${candidate[1]}\" \}")"
-  else
-      cmd_status="$(yum_status '\{ "status": "installed", "version": "%{VERSION}-%{RELEASE}" \}')" || {
-      cmd_status='{ "status": "uninstalled", "version": "" }'
-    }
-  fi
-  success "$cmd_status"
-}
-
 declare PT__installdir
-source "$PT__installdir/package/files/common.sh"
+for f in "$PT__installdir"/package/files/*sh; do
```

Comment:
Can we source these explicitly in the case statement? For example, sourced `common.sh` now, but only source `apt.sh` when `$available_manager` case matches `apt-get`.

Reply:
We could, but I don't think this would provide a lot of value.  As long as there aren't conflicting function names it doesn't matter if we `source` them all.